contrib/platform/src/com/sun/jna/platform/win32/IPHlpAPI.java

```diff
@@ -0,0 +1,280 @@
+    int MAX_INTERFACE_NAME_LEN = 256;
+    int MAXLEN_IFDESCR = 256;
+    int MAXLEN_PHYSADDR = 8;
+    int MAX_HOSTNAME_LEN = 128;
+    int MAX_DOMAIN_NAME_LEN = 128;
+    int MAX_SCOPE_ID_LEN = 256;
+
+    /**
+     * The MIB_IFROW structure stores information about a particular interface.
+     * 
+     * @see <A HREF=
+     *      "https://docs.microsoft.com/en-us/previous-versions/windows/desktop/api/ifmib/ns-ifmib-_mib_ifrow">MIB_IFROW</A>
+     */
+    @FieldOrder({ "wszName", "dwIndex", "dwType", "dwMtu", "dwSpeed", "dwPhysAddrLen", "bPhysAddr", "dwAdminStatus",
+            "dwOperStatus", "dwLastChange", "dwInOctets", "dwInUcastPkts", "dwInNUcastPkts", "dwInDiscards",
+            "dwInErrors", "dwInUnknownProtos", "dwOutOctets", "dwOutUcastPkts", "dwOutNUcastPkts", "dwOutDiscards",
+            "dwOutErrors", "dwOutQLen", "dwDescrLen", "bDescr" })
+    class MIB_IFROW extends Structure {
+        public char[] wszName = new char[MAX_INTERFACE_NAME_LEN];
+        public int dwIndex;
+        public int dwType;
+        public int dwMtu;
+        public int dwSpeed;
+        public int dwPhysAddrLen;
+        public byte[] bPhysAddr = new byte[MAXLEN_PHYSADDR];
+        public int dwAdminStatus;
+        public int dwOperStatus;
+        public int dwLastChange;
+        public int dwInOctets;
+        public int dwInUcastPkts;
+        public int dwInNUcastPkts;
+        public int dwInDiscards;
+        public int dwInErrors;
+        public int dwInUnknownProtos;
+        public int dwOutOctets;
+        public int dwOutUcastPkts;
+        public int dwOutNUcastPkts;
+        public int dwOutDiscards;
+        public int dwOutErrors;
+        public int dwOutQLen;
+        public int dwDescrLen;
+        public byte[] bDescr = new byte[MAXLEN_IFDESCR];
+    }
+
+    /**
+     * The MIB_IF_ROW2 structure stores information about a particular
+     * interface.
+     * 
+     * @see <A HREF=
+     *      "https://msdn.microsoft.com/library/windows/hardware/ff559214">MIB_IF_ROW2</A>
+     */
+    @FieldOrder({ "InterfaceLuid", "InterfaceIndex", "InterfaceGuid", "Alias", "Description", "PhysicalAddressLength",
+            "PhysicalAddress", "PermanentPhysicalAddress", "Mtu", "Type", "TunnelType", "MediaType",
+            "PhysicalMediumType", "AccessType", "DirectionType", "InterfaceAndOperStatusFlags", "OperStatus",
+            "AdminStatus", "MediaConnectState", "NetworkGuid", "ConnectionType", "TransmitLinkSpeed",
+            "ReceiveLinkSpeed", "InOctets", "InUcastPkts", "InNUcastPkts", "InDiscards", "InErrors", "InUnknownProtos",
+            "InUcastOctets", "InMulticastOctets", "InBroadcastOctets", "OutOctets", "OutUcastPkts", "OutNUcastPkts",
+            "OutDiscards", "OutErrors", "OutUcastOctets", "OutMulticastOctets", "OutBroadcastOctets", "OutQLen" })
+    class MIB_IF_ROW2 extends Structure {
+        public long InterfaceLuid; // 64-bit union
+        public ULONG InterfaceIndex;

```

Comment:
Would int be an alternative? (applies to other definitions as well).

Reply:
Also applies to some of my other PRs.  As a structure mapping it would work, however, I have tried (not always consistently) to use the mapped type when variables are unsigned.  I've gone back and forth a few times on many of these... using int, we require the user to know to interpret the unsigned value, while ULONG or DWORD offer `longvalue()` options.

Reply:
It's ok to leave it as int here, because the javadoc makes it clear what the underlying value is and there's likely a tiny performance improvement to avoiding the extra object creation.  As I said I went back and forth on this (I literally did search-and-replace, and then undo, at least twice) before submitting.   

I might suggest a Global JNA util method that converts unsigned values, though, so users don't have to always reinvent the wheel.